sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -2369,6 +2369,23 @@ interface WithAdditionalCapacities {
              */
             Update disableHibernation();
         }
+
+        /** The stage of the VM update allowing to swap os disk. */
+        interface SwapOSDisk {
+            /**
+             * Swaps the OS disk of the virtual machine for the specified managed disk.
+             * <p>Note:</p>
+             * <p>- Both the original OS disk and the disk to swap for have to be managed disks.</p>
+             * <p>- You cannot switch the OS Type of the VM
+             *      (i.e. Swap an OS Disk with Linux for an OS Disk with Windows).</p>
+             * <p>- Make sure that the VM size and storage type are compatible with the disk you want to attach.
+             *      For example, if the disk you want to use is in Premium Storage, then the VM needs to be capable of
+             *      Premium Storage (like a DS-series size).</p>
+             * @param diskId id of the managed disk to swap OS disk for
+             * @return the next stage of the update
+             */
+            Update swapOSDisk(String diskId);
```

Comment:
I am not sure what to call this, will `withOSDisk` (and in javadoc say it is a replacement, not an add) do?

Also I assume we should have an overload with `Disk disk`?

Reply:
Sounds good to me. For javadoc, maybe current will do, since "swap" is a good indication that it's not an add?

Reply:
Sure. Javadoc is fine. Even if we change, nothing would affect API.